pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,47 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it won't stop until a cancel signal happens.
+func (p *Puller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) error {
+	var rErr error
+	tailClient, err := p.client.TailLogEntries(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create tailLogEntriesClient: %w", err)
+	}
+
+	defer func() {
+		if closeClientErr := tailClient.CloseSend(); closeClientErr != nil {
+			rErr = errors.Join(rErr, fmt.Errorf("failed to close tailClient %w", closeClientErr))
```

Comment:
I don't think the error is ever returned.

Reply:
I tried in here: https://go.dev/play/p/x0pcIa73OVw

Reply:
Updated.